backend/src/core/middlewares/campaign.middleware.ts

```diff
@@ -10,15 +15,18 @@ import { CampaignService, TemplateService } from '@core/services'
 const canEditCampaign = async (
   req: Request,
   res: Response,
-  next: NextFunction
+  next: NextFunction,
+  protect = false
 ): Promise<Response | void> => {
   try {
     const { campaignId } = req.params
-    const [hasJob, csvStatus] = await Promise.all([
+    const [hasJob, csvStatus, isProtected] = await Promise.all([
       CampaignService.hasJobInProgress(+campaignId),
-      TemplateService.getCsvStatus(+campaignId),
+      UploadService.getCsvStatus(+campaignId),
+      ProtectedService.isProtectedCampaign(+campaignId),
     ])
-    if (!hasJob && !csvStatus?.isCsvProcessing) {
+    if (!hasJob && !csvStatus?.isCsvProcessing && (isProtected || !protect)) {
```

Comment:
All email campaign routes are available to a protected email campaign, but protected campaign routes are only available to protected email campaigns, right?

If my assumption above was correct, then the check  `ProtectedService.isProtectedCampaign` should be extracted into separate middleware, to avoid making this query for normal email campaigns.
Eg.
```javascript
router.get(
  '/protect/upload/start',
  celebrate(startMultipartValidator),
  CampaignMiddleware.canEditProtectedCampaign,
  UploadMiddleware.startMultipartUpload
)
``` 
would become
```javascript
// CampaignMiddleware.isProtectedCampaign
 isProtectedCampaign = (req, res, next) => { 
   if(!ProtectedService.isProtectedCampaign(+req.params.campaignId)) return res.sendStatus(403) 
  },

router.get(
  '/protect/upload/start',
  celebrate(startMultipartValidator),
  CampaignMiddleware.canEditCampaign,
  CampaignMiddleware.isProtectedCampaign,
  UploadMiddleware.startMultipartUpload
)
```